app/src/main/java/com/iic/shopingo/ui/request_flow/RequestStateActivity.java

```diff
@@ -0,0 +1,48 @@
+package com.iic.shopingo.ui.request_flow;
+
+import android.os.Bundle;
+import android.support.v7.app.ActionBarActivity;
+import android.view.View;
+import com.iic.shopingo.R;
+
+public class RequestStateActivity extends ActionBarActivity {
+
+  public static final String REQUEST_STATE_EXTRA_KEY = "request_state";
+
+  @Override
+  protected void onCreate(Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+    RequestStatus state = RequestStatus.values()[getIntent().getIntExtra(REQUEST_STATE_EXTRA_KEY, 0)];
+    int layout;
+    switch (state) {
+      case APPROVED:
+        layout = R.layout.activity_request_state_approved;
+        break;
+      case DECLINED:
+        layout = R.layout.activity_request_state_declined;
+        break;
+      default:
+        layout = R.layout.activity_request_state_pending;
+    }
+    setContentView(layout);
+  }
+
+  public void onCancelRequest(View view) {
```

Comment:
Oh, you did it in the XML.
I don't really like that. It makes the layouts harder to reuse and the callbacks harder to find.